### PR TITLE
cpu/atmega_common

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

- Minor cleanups in `cpu/atmega_common`:
    - No longer using [reserved identifiers](https://wiki.sei.cmu.edu/confluence/display/c/DCL37-C.+Do+not+declare+or+define+a+reserved+identifier)
    - ~~Use a C11 atomic instead of `volatile` + `__asm__ volatile("" ::: "memory");` for the flag used to check if the ATmega is in IRQ context~~
- Reworked the code that is run at the end of an ISR
    - The current code behaved like `thread_yield()` is executed at the end of an ISR, but correct behavior is `thread_yield_higher()`; This PR changes this to behave like `thread_yield_higher()`
    - With this, `tests/isr_yield_higher` passes
    - This also save some lines of code an ROM :-)

### Testing procedure

1. Confirm that `tests/isr_yield_higher` now really passes
2. Run a bunch of other tests related to context switching, scheduling and interrupts to verify that no regressions are introduced. (E.g. `tests/thread_cooperation/`, `tests/thread_basic/`, `tests/periph_timer`, `tests/sched_testing`, ...)

### Issues/PRs references

Tick `tests/isr_yield_higher` in https://github.com/RIOT-OS/RIOT/issues/12651

### CODE REVIEW

File: cpu/atmega_common/irq_arch.c [hunk 1]
```diff
@@ -29,12 +29,12 @@
 /**
  * @brief Macro returns state of the global interrupt register
  */
-static uint8_t __get_interrupt_state(void);
-static void __set_interrupt_state(uint8_t state);
+static uint8_t get_interrupt_state(void);
```
Comment (aabadie) [inline cpu/atmega_common/irq_arch.c]: Maybe also namespace these 2 functions with `atmega_` ?

File: cpu/atmega_common/thread_arch.c [hunk 2]
```diff
@@ -34,7 +34,7 @@
  * local function declarations  (prefixed with __)
  */
 static void __context_save(void);
-static void __context_restore(void);
+static void atmega_context_restore(void);
```
Comment (aabadie) [inline cpu/atmega_common/thread_arch.c]: Because of consistency, we can't leave one function with namespace `atmega_` and not the others.

Comment (maribu) [reply cpu/atmega_common/thread_arch.c]: I somehow overlooked the others. Thanks for spotting this!

File: cpu/atmega_common/include/cpu.h [hunk 3]
```diff
@@ -34,6 +34,8 @@
 #include <stdint.h>
 
 #include <avr/interrupt.h>
+#include <stdint.h>
```
[the remark this answers is not in the document]
Comment (maribu) [reply cpu/atmega_common/include/cpu.h]: Fixed

File: cpu/atmega_common/irq_arch.c (hunk 1 above)
Comment (maribu) [reply cpu/atmega_common/irq_arch.c]: Fixed

**Comment by maribu** [comment]
Using the C11 atomics is inherently incompatible with C++. I was aware of that, but I wasn't aware that `cpu.h` also gets included from C++.

I could use the preprocessor to only provide the `atmega_enter_irq()` and `atmega_exit_irq()` only to C. Or move the `static inline atmega_enter_isr()` into a C file, so that the `#include <stdatomic>` from `cpu.h` could be dropped. But I now I'm no longer sure if this really is an improvement in readability.

I just revert the change to C11 atomics now, so that the actual bugfix can get in swiftly.

**Comment by maribu** [comment]
OK. I updated to remove the C11 atomics and squash. The commits were a bit mere entangled that I expected, so I had to do some manual merging. I hope I didn't mess up anything :-D

**Comment by maribu** [comment]
Murdock is super helpful of detecting all the leftovers :-)

The fact that atmega specific code is called in `drviers/at86rf2xx` only confirms that the change in names to add the `atmega_` prefix is making this more readable.

**Comment by maribu** [comment]
@kaspar030: The Murdock failure seems unrelated, but nasty. Like a bug that is only triggered when the stars are aligned in a specific way. Murdock doesn't provide core dumps for tests failed on native, does it?

**Comment by maribu** [comment]
I triggered a rebuild now. The issue [is documented here](https://github.com/RIOT-OS/RIOT/issues/12796), in case this is needed. Because a second similar issue occurred on the same machine, I now think that too much load resulting in the timeouts in both cases is the most plausible explanation, rather than a latent bug locking up RIOT on `native` every now and then.

**Pull request merged**